### PR TITLE
updates to arch --> core concepts

### PR DESCRIPTION
**Pull request by tpoitras**
as per docs trello card: https://trello.com/c/K3GLjId7

### CODE REVIEW

**Comment by adellape** [comment]
cc @sosiouxme

File: architecture/core_concepts/overview.adoc [hunk 1]
```diff
@@ -7,10 +7,8 @@
 
 The following topics provide high-level, architectural information on core
 concepts and objects you will encounter when using OpenShift. Many of these
-objects come from link:../../rest_api/kubernetes_v1.html[the model provided by
-Kubernetes], which is link:../../rest_api/openshift_v1.html[extended by
-OpenShift's model] to provide a more feature-rich development lifecycle
-platform.
```
Comment (sosiouxme) [inline architecture/core_concepts/overview.adoc]: Any particular rationale behind removing these links to the REST API? They're not very overview-y but I thought a quick link to the nuts and bolts of k8s vs openshift was germane.

**Comment by sosiouxme** [comment]
Looked through, looks like good improvements, thanks!

File: architecture/core_concepts/overview.adoc (hunk 1 above)
Comment (adellape) [inline architecture/core_concepts/overview.adoc]: @tpoitras Wanted to get your feedback on this before merging.

Comment (tpoitras) [inline architecture/core_concepts/overview.adoc]: Hey @sosiouxme -- It's just not the type of content I think we should be linking to from an overview. As you said, it's not very overview-y. I reckon if someone wants to check out the nuts & bolts, the API content is pretty top-level and easy to find in the docs. :cactus:

**Comment by tpoitras** [comment]
Thank you, Luke. I've commented above. 

Alex, can merge if all good now? :ghost:

**Pull request closed without merging**